src/__tests__/lib/containers/entity/annotations/AdditionalPropertiesSchemaField.test.ts

```diff
@@ -45,6 +49,25 @@ describe('AdditionalPropertiesSchemaField unit tests', () => {
       ])
     })
 
+    it('converts integers to floats', () => {
+      const data = [1, 2, 3]
+      expect(transformDataFromPropertyType(data, PropertyType.FLOAT)).toEqual([
+        '1.0',
+        '2.0',
+        '3.0',
+      ])
+    })
+
+    it('converts strings to floats', () => {
+      const data = ['abc', '3.4', '4', 'NaN']
+      expect(transformDataFromPropertyType(data, PropertyType.FLOAT)).toEqual([
+        'NaN',
+        3.4,
+        '4.0',
```

Comment:
Why does the '3.4' input get transformed to 3.4, but '4' is a string ('4.0')?   I guess I expected it to be 4.0

Reply:
I switched the implementation from appending `".0"` to using `Number.toFixed`, but in either case, the output is a string. We have to use a string because the number `4.0` is automatically converted to `4`